### PR TITLE
fix: addressing some unanticipated situations

### PR DESCRIPTION
**Pull request by idanran**
Avoid requesting unnecessary permissions when used by Deno users.
Modern Chromium supports true color.

### CODE REVIEW

**Comment by sindresorhus** [comment]
CI is failing and this PR needs a proper title.

**Comment by idanran** [comment]
> CI is failing and this PR needs a proper title.

Okay, I'll try to modify.

**Comment by idanran** [comment]
> CI is failing and this PR needs a proper title.

done

File: index.js [hunk 1]
```diff
@@ -84,6 +84,10 @@ function _supportsColor(haveStream, {streamIsTTY, sniffFlags = true} = {}) {
 		return 0;
 	}
 
+	if (env.COLORTERM === 'truecolor') {
+		return 3;
+	}
```
Comment (sindresorhus) [inline index.js:89]: Why is this moved?

File: browser.js [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+		hasBasic: true,
+		has256: level >= 2,
+		has16m: level >= 3,
+	};
+}
+
+function check() {
+	if (navigator.userAgentData) {
+		const brand = navigator.userAgentData.brands.find(({brand}) => brand === 'Chromium');
+		if (brand && brand.version > 93) {
```
Comment (sindresorhus) [inline browser.js:19]:
```suggestion
		if (brand?.version > 93) {
```

File: index.js (hunk 1 above)
Comment (idanran) [reply index.js:89]: > Why is this moved?

COLORTERM is present in most cases. And it reduces the number of times Deno needs to obtain permissions for every env it accesses. Deno has to request permission once for each access to an env.

File: browser.js [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+	}
+
+	if (/\b(Chrome|Chromium)\//.test(navigator.userAgent)) {
+		return 1;
+	}
+
+	return 0;
+}
+
+const colorSupport = translateLevel(check());
```
Comment (Qix-) [inline browser.js:31]: The call to check can be an IIFE.

File: index.js (hunk 1 above)
Comment (Qix-) [reply index.js:89]: This technically breaks our API...

**Review by Qix- — CHANGES_REQUESTED** [review]
I'm not particularly fond of these changes but with the requested change it isn't the worst I suppose.

Maybe I'm missing something (your PR doesn't explain what these changes actually do...) but what are the changes in the browser implementation trying to achieve?

File: browser.js (hunk 3 above)
Comment (idanran) [reply browser.js]: done

**Comment by idanran** [comment]
> I'm not particularly fond of these changes but with the requested change it isn't the worst I suppose.
> 
> Maybe I'm missing something (your PR doesn't explain what these changes actually do...) but what are the changes in the browser implementation trying to achieve?

Starting with the specified version, Chromium's console can output true color.

File: index.js (hunk 1 above)
Comment (idanran) [reply index.js]: Restored

**Review by Qix- — APPROVED** [review]
These look better.

File: index.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import os from 'node:os';
 import tty from 'node:tty';
 
 // From: https://github.com/sindresorhus/has-flag/blob/main/index.js
-function hasFlag(flag, argv = process.argv) {
+function hasFlag(flag, argv = globalThis.Deno?.args || process.argv) {
```
Comment (sindresorhus) [inline index.js:6]:
```suggestion
function hasFlag(flag, argv = globalThis.Deno?.args ?? process.argv) {
```
